src/main/java/edu/kit/provideq/toolbox/ProcessRunner.java

```diff
@@ -1,40 +1,220 @@
 package edu.kit.provideq.toolbox;
 
+import edu.kit.provideq.toolbox.meta.ProblemType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
 import java.io.BufferedReader;
+import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
 
 /**
- * Runs a process.
+ * Runs a process to solve a problem.
+ * Accepts a ProblemBuilder
  * Is meant to be expanded with the builder pattern.
  */
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 public class ProcessRunner {
```

Comment:
Do you think we should separate the builder pattern from the process runner? I'm not 100% sure but integrating the builder pattern into the thing that is only supposed to use the built result might be confusing (e.g., all of these configuration methods are exposed from `GamsProcessRunner` too, do we want that?).

Reply:
I developed this with having GAMS scripts in mind, that might need other I/O files than the one we specified.
While you don't need it, you could still use `GamsProcessRunner` without formatting the script to use our standardized I/O handling while keeping the GAMS specific execution with obfuscating the license etc.

```java
var processResult = context
                .getBean(
                        GamsProcessRunner.class,
                        maxCutPath,
                        "maxcut.gms")
                .addProblemFilePathToProcessCommand("--SPECIALINPUT=\"%s\"")
                .addSolutionFilePathToProcessCommand()
                .solutionFileName("sol.xyz")
                .run(problem.type(), solution.getId(), problem.problemData())
```